lib/lz4.c

```diff
@@ -1852,19 +1856,19 @@ LZ4_decompress_generic(
             {
                 if (partialDecoding) {
                     if (cpy > oend) { cpy = oend; assert(op<=oend); length = (size_t)(oend-op); }  /* Partial decoding : stop in the middle of literal segment */
-                    if ((endOnInput) && (ip+length > iend)) { goto _output_error; } /* Error : read attempt beyond end of input buffer */
+                    if ((endOnInput) && (ip+length>iend)) { goto _output_error; } /* Error : read attempt beyond end of input buffer */
+                    if ((endOnInput) && (cpy != oend) && (ip+length>iend-(2+1+LASTLITERALS)) && (ip+length != iend)) { goto _output_error; } /* Error : input must be consumed */
```

Comment:
I believe this is correct, but maybe it could be written better? Please carefully check this fix, since I'm not 100% confident in it.

It passes the round trip and decompression fuzzers, which gives me some confidence that its right.